ToneMapper/Model/Infastracture/TMGLEngine.m

```diff
@@ -71,10 +75,19 @@ - (void)draw {
               withProgramParameters:self.workspaceInspectorProgram];
 }
 
-- (void)useFilter:(id<TMGLFilter> __nonnull)filter {
+- (void)applyFilter:(id<TMGLFilter> __nonnull)filter {
```

Comment:
Use <code>NS_ASSUME_NONULL_BEGIN / END</code> instead of <code>__nonnull</code>.